### PR TITLE
tests: begin cleaning up the Macro tests for Windows

### PR DESCRIPTION
**Pull request by compnerd**
[unsure still]

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @rintaro @bnbarham @DougGregor

File: test/lit.cfg [hunk 1]
```diff
@@ -531,7 +531,7 @@ shutil.rmtree(completion_cache_path, ignore_errors=True)
 ccp_opt = "-completion-cache-path %r" % completion_cache_path
 lit_config.note("Using code completion cache: " + completion_cache_path)
 
-config.swift_host_lib_dir = make_path(config.swift_lib_dir, 'swift', 'host')
+config.swift_host_lib_dir = config.swift_bin_dir
```
Comment (rintaro) [inline test/lit.cfg:534]: Could you add `%swift-host-plugin-dir`?
For Darwin/Linux/etc. it'd be `make_path(config.swift_lib_dir, 'swift', 'host', 'plugins')`
For Windows, `config.swift_bin_dir`

Comment (compnerd) [reply test/lit.cfg:534]: Yeah, except I think that we should extract that change.

**Comment by compnerd** [comment]
Closing this off as all the various changes to the tests are split up into separate PRs and this was just tracking the set of changes that we needed.

**Pull request closed without merging**